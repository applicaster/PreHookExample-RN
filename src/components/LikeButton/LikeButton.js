import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { sendAnalyticEvent } from '@applicaster/react-native-zapp-bridge';
import { FB_LIKE_CLICKED } from '../../constants/analyticEvents';
import ActionButton from '../ActionButton';
import { LIKE_BUTTON } from '../../icons';

class LikeButton extends Component {
  constructor(props) {
    super(props);
    this.navigateToWebview = this.navigateToWebview.bind(this);
  }

  navigateToWebview() {
    const { navigation } = this.context;
    const { eventId, setActiveEventId } = this.props;

    setActiveEventId(eventId);
    navigation.navigate('SocialWebView', { headerTitle: 'Facebook' });
    sendAnalyticEvent(FB_LIKE_CLICKED, { eventId }).then().catch();
  }
  
  render() {
    const { likesCount } = this.props;
    return (
      <ActionButton
        imageUri={LIKE_BUTTON}
        label={likesCount}
        onPress={this.navigateToWebview}
      />
    );
  }
}

LikeButton.propTypes = {
  eventId: PropTypes.string,
  likesCount: PropTypes.number,
  setActiveEventId: PropTypes.func,
};

LikeButton.contextTypes = {
  navigation: PropTypes.object,
};

export default LikeButton;
