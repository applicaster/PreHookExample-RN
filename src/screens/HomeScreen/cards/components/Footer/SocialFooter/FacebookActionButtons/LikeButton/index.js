import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setActiveEventId } from '../../../../../../../../actions';

import LikeButton from './LikeButton';

const mapDispatchToProps = dispatch => bindActionCreators({
  setActiveEventId,
}, dispatch);

export default connect(undefined, mapDispatchToProps)(LikeButton);
