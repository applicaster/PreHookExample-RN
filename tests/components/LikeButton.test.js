import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LikeButton from '../../src/components/LikeButton/LikeButton';

jest.mock('react-native-zapp-bridge', () => 'NativeZappBridge');

describe('LikeButton', () => {
  test('component renders correctly', () => {
    const tree = renderer.create(
      <LikeButton likesCount={3} eventId={1} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
