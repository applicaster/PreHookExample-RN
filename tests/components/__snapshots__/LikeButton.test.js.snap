// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "marginRight": 7,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    collapsable={undefined}
    source={
      Object {
        "uri": "feed_rn_like_button",
      }
    }
    style={
      Object {
        "height": 30,
        "marginRight": 1,
        "tintColor": "#FFFFFF",
        "transform": Array [
          Object {
            "scaleX": 1,
          },
          Object {
            "scaleY": 1,
          },
        ],
        "width": 30,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  >
    3
  </Text>
</View>
`;
